staging/src/k8s.io/client-go/tools/dnscache/dialer.go

```diff
@@ -0,0 +1,283 @@
+package dnscache
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/golang/glog"
+)
+
+// Dialer describes a net dialer. This interface is an abstract of net.Dialer.
```

Comment:
Don't redeclare net.Dialer?

Reply:
`net.Dialer` and `net.Resolver` are structs. So I need to define a interface for net.Dialer.

Reply:
Ah, sorry, I glanced at the package but not closely enough, it's been a bit since I last did something with a custom resolver.